src/modules/auth/auth.service.ts

```diff
@@ -38,26 +44,73 @@ export class AuthService {
     const { ...result } = user;
     delete result.password;
 
-    const authToken = await this.generateAuthToken({
-      sub: user.id,
-      userEmail: user.email,
-      userRole: user.role,
-    });
+    const authToken = await this.generateAuthToken(
+      {
+        sub: user.id,
+        userEmail: user.email,
+        userRole: user.role,
+      },
+      'access',
+    );
     const data = {
       result: result,
       token: authToken,
     };
 
     return data;
   }
-  async generateAuthToken(payload: AuthTokenPayload) {
-    const secret = this.configService.get('JWT_ACCESS_TOKEN_SECRET', {
-      infer: true,
-    });
-    const expiresIn = this.configService.get('JWT_ACCESS_TOKEN_EXPIRY', {
-      infer: true,
+
+  async login({ email, password }: LoginDto) {
+    const user = await this.userRepository.findOne({
+      where: { email },
     });
 
+    if (!user) {
+      throw new BadRequestException('Invalid email or password');
+    }
+
+    const isPasswordValid = await Hasher.verifyHash(user.password, password);
+    if (!isPasswordValid) {
+      throw new BadRequestException('Invalid email or password');
+    }
+
+    const payload = {
+      sub: user.id,
+      userEmail: user.email,
+      userRole: user.role as Roles,
```

Comment:
Why are you using type assertion? I think user.role itself should be of type Roles